oauth2-authorization-server/src/test/java/org/springframework/security/config/annotation/web/configurers/oauth2/server/authorization/OidcUserInfoTests.java

```diff
@@ -0,0 +1,255 @@
+import org.springframework.security.web.util.matcher.RequestMatcher;
+import org.springframework.test.web.servlet.MockMvc;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+/**
+ * Integration tests for the OpenID User Info endpoint.
+ *
+ * @author Steve Riesenberg
+ */
+public class OidcUserInfoTests {
+	private static final String DEFAULT_OIDC_USER_INFO_ENDPOINT_URI = "/userinfo";
+	private static final List<String> OPENID_USER_INFO_SCOPES = Arrays.asList(
+			"openid", "address", "email", "phone", "profile"
+	);
+
+	@Rule
+	public final SpringTestRule spring = new SpringTestRule();
+
+	@Autowired
+	private MockMvc mvc;
+
+	@Autowired
+	private JwtEncoder jwtEncoder;
+
+	@Autowired
+	private OAuth2AuthorizationService authorizationService;
+
+	@Test
+	public void requestWhenUserInfoRequestGetThenUserInfoResponse() throws Exception {
+		this.spring.register(AuthorizationServerConfiguration.class).autowire();
+
+		OAuth2Authorization authorization = createAuthorization();
+		this.authorizationService.save(authorization);
+
+		// @formatter:off
+		this.mvc.perform(get(DEFAULT_OIDC_USER_INFO_ENDPOINT_URI)
+				.header(HttpHeaders.AUTHORIZATION, "Bearer " + authorization.getAccessToken().getToken().getTokenValue()))
+				.andExpect(status().is2xxSuccessful())
+				.andExpect(jsonPath("sub").value("user1"))
+				.andExpect(jsonPath("name").value("First Last"))
+				.andExpect(jsonPath("given_name").value("First"))
+				.andExpect(jsonPath("family_name").value("Last"))
+				.andExpect(jsonPath("middle_name").value("Middle"))
+				.andExpect(jsonPath("nickname").value("User"))
+				.andExpect(jsonPath("preferred_username").value("user"))
+				.andExpect(jsonPath("profile").value("https://example.com/user1"))
+				.andExpect(jsonPath("picture").value("https://example.com/user1.jpg"))
+				.andExpect(jsonPath("website").value("https://example.com"))
+				.andExpect(jsonPath("email").value("user1@example.com"))
+				.andExpect(jsonPath("email_verified").value("true"))
+				.andExpect(jsonPath("gender").value("female"))
+				.andExpect(jsonPath("birthdate").value("1970-01-01"))
+				.andExpect(jsonPath("zoneinfo").value("Europe/Paris"))
+				.andExpect(jsonPath("locale").value("en-US"))
+				.andExpect(jsonPath("phone_number").value("+1 (604) 555-1234;ext=5678"))
+				.andExpect(jsonPath("phone_number_verified").value("false"))
+				.andExpect(jsonPath("address").value("Champ de Mars\n5 Av. Anatole France\n75007 Paris\nFrance"))
+				.andExpect(jsonPath("updated_at").value("1970-01-01T00:00:00Z"));
+		// @formatter:on
+	}
+
+	@Test
+	public void requestWhenUserInfoRequestPostThenUserInfoResponse() throws Exception {
```

Comment:
Let's try to share some code between the GET and POST test methods to avoid duplication.